uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/AbsenceType.tpl

```diff
@@ -270,11 +274,58 @@
                       {literal}
                   }
 
+                function initCalculationUnitControls() {
+                  setEntitlementLabelText();
+                  var prev_calculation_unit_val = $('.absence-calculation-unit select option:selected').val();
+                  $('.absence-calculation-unit select').on('change', function(e) {
+                    var add_public_holiday_entitlement = $('.public-holiday-option input[type=radio]:checked').val();
+                    var hours_unit_value = getHoursUnitValue();
+                    if(this.value == hours_unit_value && add_public_holiday_entitlement == 1) {
+                      $('.absence-calculation-unit select').select2('val', prev_calculation_unit_val);
+                      showPublicHolidayEntitlementErrorAlert();
+                    }
+                    else{
+                      prev_calculation_unit_val = $('.absence-calculation-unit select option:selected').val();
+                    }
+                    setEntitlementLabelText();
+                  });
+                }
+
+                function setEntitlementLabelText() {
+                  $('.entitlement-label').text($('.absence-calculation-unit select option:selected').text());
+                }
+
+                function initPublicHolidayControls() {
+                  $('.public-holiday-option input[type=radio]').on('click', function(e) {
+                    var calculation_unit = $('.absence-calculation-unit select option:selected').val();
+                    var hours_unit_value = getHoursUnitValue();
+                    if (this.value == 1 && calculation_unit == hours_unit_value) {
+                      showPublicHolidayEntitlementErrorAlert();
+                      e.preventDefault();
+                    }
+                  });
+                }
+
+                function showPublicHolidayEntitlementErrorAlert() {
+                  CRM.alert("You cannot add public holiday to entitlement when calculation unit is in Hours",
+                    'Modify Absence Type', 'error');
+                }
+
+                function getHoursUnitValue() {
+                  {/literal}
+                  var hours_unit_value = {$hoursUnitValue};
+                  {literal}
+
+                  return hours_unit_value;
+                }
+
                   $(document).ready(function() {
                       initToilControls();
                       initCarryForwardControls();
                       initColorPicker();
                       initDeleteButton();
+                      initCalculationUnitControls()
+                      initPublicHolidayControls()
```

Comment:
It looks like that some of the old code here is still using 4 spaces instead of 2 for indentation. Can you please fix that as well?